### ISSUE TITLE
yum install failed with AttributeError: \"'NoneType' object has no attribute 'unshare_data

### ISSUE THREAD

**Issue by ansibot** [ansible/ansible#29686]
From @azureliang on 2016-09-01T09:57:39Z

##### Issue Type
    Bug Report

##### Component Name
yum module

##### Ansible Version
2.3

##### Environment

Ansible Master: Oracle Linux 6 
Remote Hosts: Oracle Linux 5.11

##### Summary

I met following error when do yum install , seems the package is already installed, but an exception occured:
```
tasks:
- name: Yum install dependency packages
  environment:
    http_proxy: http://rmdc-proxy.oracle.com:80
  yum:
    name: "{{ item }}"
    state:  "installed"
  with_items:
  - sudo

```
Trace : 
```
Error:failed: [10.252.155.73] => (item=[u'sudo']) => {"failed": true, "invocation": {"module_name": "yum"}, "item": ["sudo"], "module_stderr": "OpenSSH_5.3p1, OpenSSL 1.0.1e-fips 11 Feb 2013\ndebug1: Reading configuration data /etc/ssh/ssh_config\r\ndebug1: Applying options for *\r\ndebug1: auto-mux: Trying existing master\r\ndebug2: fd 3 setting O_NONBLOCK\r\ndebug2: mux_client_hello_exchange: master version 4\r\ndebug3: mux_client_request_forwards: requesting forwardings: 0 local, 0 remote\r\ndebug3: mux_client_request_session: entering\r\ndebug3: mux_client_request_alive: entering\r\ndebug3: mux_client_request_alive: done pid = 295\r\ndebug3: mux_client_request_session: session request sent\r\ndebug1: mux_client_request_session: master session id: 2\r\ndebug1: mux_client_request_session: master session id: 2\r\ndebug3: mux_client_read_packet: read header failed: Broken pipe\r\ndebug2: Received exit status from master 0\r\nShared connection to 10.252.155.73 closed.\r\n", "module_stdout": "{\"msg\": \"\", \"invocation\": {\"module_args\": {\"name\": [\"sudo\"], \"list\": null, \"install_repoquery\": true, \"conf_file\": null, \"disable_gpg_check\": false, \"state\": \"installed\", \"disablerepo\": null, \"update_cache\": false, \"enablerepo\": null, \"exclude\": null}}, \"changed\": true, \"results\": [\"Loaded plugins: security\nExcluding Packages in global exclude list\nFinished\nSetting up Install Process\nResolving Dependencies\n--> Running transaction check\n---> Package sudo.x86_64 0:1.7.2p1-29.el5_10 set to be updated\n--> Finished Dependency Resolution\n\nDependencies Resolved\n\n================================================================================\n Package      Arch           Version                   Repository          Size\n================================================================================\nInstalling:\n sudo         x86_64         1.7.2p1-29.el5_10         el5_latest         364 k\n\nTransaction Summary\n================================================================================\nInstall       1 Package(s)\nUpgrade       0 Package(s)\n\nTotal download size: 364 k\nDownloading Packages:\nRunning rpm_check_debug\nRunning Transaction Test\nFinished Transaction Test\nTransaction Test Succeeded\nRunning Transaction\n\r  Installing     : sudo                                                     1/1 \nwarning: /etc/sudoers created as /etc/sudoers.rpmnew\n\nInstalled:\n  sudo.x86_64 0:1.7.2p1-29.el5_10                                               \n\nComplete!\n\"], \"rc\": 0}\r\nException exceptions.AttributeError: \"'NoneType' object has no attribute 'unshare_data'\" in <bound method YumBase.__del__ of <yum.YumBase object at 0x1d77d10>> ignored\r\n", "msg": "MODULE FAILURE", "parsed": false}
```
Copied from original issue: ansible/ansible-modules-core#4631

**Comment by ansibot** [comment]
From @ansibot on 2016-09-01T09:57:39Z

@azureliang, we need more information to resolve this issue. For that reason, we put this Issue into the 'needs_info' state.

Here are the required items we could not find in your description:
- issue type

When you have filled in the missing data, we will notify the module maintainer for further action.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
From @ansibot on 2016-09-01T09:57:39Z

@ansible, ping. This issue is waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@ansibot Greetings! Thanks for taking the time to open this issue. In order for the community to handle your issue effectively, we need a bit more information. 

Here are the items we could not find in your description:
- ansible version
- component name

Please set the description of this issue with this template:
https://raw.githubusercontent.com/ansible/ansible/devel/.github/ISSUE_TEMPLATE.md

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @berenddeschouwer @verm666
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @mkrizek
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @akasurde
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by maxamillion** [comment]
Thank you for the issue report! I am currently going through the yum module bug backlog and I noticed this is reported against Oracle Linux 5.11 as the remote host, which is not supported by Ansible. However, if you can reproduce this with a remote host of OL 6.x I would be happy to look into it further.

needs_info

**Comment by ansibot** [comment]
cc @kustodian
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@azureliang This issue is waiting for your response. Please respond or the issue will be closed.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@azureliang You have not responded to information requests in this issue so we will assume it no longer affects you. If you are still interested in this, please create a new issue with the requested information.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)
